docs/apiref/firmware.rst

```diff
@@ -132,3 +136,88 @@ To remove a firmware image:
     "job_id": "5d849177dd428720db72c693"
    }
     
+
+Upgrade firmware on device(s)
+-----------------------------
+
+As of today we support upgrading firmware on Arista EOS acces
+switches. The upgrade procedure can do a 'pre-flight check' which will
+make sure there is enough disk space before attempting to download the
+firmware.
+
+The API method will accept a few parameters:
+
+* group: The name of a group, all devices in that group will be upgraded.
+* hostname: If a hostname is specified, this single device will be upgraded.
+* filename: Name of the new firmware, for example "test.swi".
+* url: URL to the firmware storage, for example "http://hostname/firmware/". This should typically point to the CNaaS NMS server and files will be downloaded from the CNaaS HTTP server.
+* download: Only download the firmware.
```

Comment:
It would be nice to show the default behaviour for these options, or if they are required to be specified

Reply:
Clarified.